ocaml/typing/typecore.ml

```diff
@@ -5430,9 +4267,8 @@ and type_tuple_approx (env: Env.t) loc ty_expected l =
 
 and type_approx_function =
   let rec loop env params c body ty_expected ~in_function ~first =
-    let open Jane_syntax.N_ary_functions in
     let loc_function, _ = in_function in
-    let loc = loc_rest_of_function ~loc_function params body in
+    let loc = loc_rest_of_function ~loc_function ~first params body in
     (* We can approximate types up to the first newtype parameter, whereupon
       we give up.
     *)
```

Comment:
I don't think we've dropped that &mdash; I think `filter_arrow` handles that case for us. I think this will be a bit hard to puzzle through as-is, so I recommend merging for now and then seeing if the test case that Stephen added fails. (`ocaml/testsuite/tests/typing-misc/let_rec_approx.ml`. I confirm we have it.)